src/compute/src/sink/copy_to_s3_oneshot.rs

```diff
@@ -77,18 +80,36 @@ where
             "Consolidated COPY TO S3 input",
         );
 
+        // We need to consolidate the error collection to ensure we don't act on retracted errors.
         let error = consolidate_pact::<KeyBatcher<_, _, _>, _, _, _, _, _>(
-            &err_collection.map(move |row| {
-                let batch = row.hashed() % batch_count;
-                ((row, batch), ())
-            }),
-            Exchange::new(move |(((_, batch), _), _, _)| *batch),
-            "Consolidated COPY TO S3 error",
+            &err_collection.map(|row| (row, ())),
+            Pipeline,
```

Comment:
I think you might still need an exchange here because there's no guarantee that errors will cancel each other locally.

Reply:
ah yes, good point!